### PR TITLE
Validation for response in ff58

### LINKED ISSUE #139: Leadfoot throws TypeError when trying to launch new FF 58 with GeckoDriver 19.1
Environment:

Mac OSX 10.3.3
Node.js - 6.12.2
Leadfoot: 1.7.4  - 2.0.0 beta 8
Browser: Firefox 58
GeckoDriver: 19.1

1.) Use leadfoot directly to start browser
2.) Use standard capabilities

Expected Result:
Browser starts with capabilities

Actual Result: 
Browser fails to start with stack trace:

RROR  TypeError: Cannot read property 'platform' of undefined
           at Server._fillCapabilities (/Users/silne30/Desktop/kibana/node_modules/leadfoot/Server.js:395:19)
           at /Users/silne30/Desktop/kibana/test/functional/services/remote/leadfoot_command.js:33:18
           at undefined.next (native)
           at step (/Users/silne30/Desktop/kibana/test/functional/services/remote/leadfoot_command.js:104:191)
           at /Users/silne30/Desktop/kibana/test/functional/services/remote/leadfoot_command.js:104:361
           at process._tickDomainCallback (internal/process/next_tick.js:135:7)

### PR DESCRIPTION
**Pull request by silne30**
Fixes #139

### CODE REVIEW

**Comment by silne30** [comment]
@dylans Yep. One commit was in the wrong email. My bad. Thanks for the help.

**Comment by dylans** [comment]
Reopening as it hasn't landed in Leadfoot yet.

File: Server.js [hunk 1]
```diff
@@ -363,12 +363,19 @@ Server.prototype = {
 			desiredCapabilities: desiredCapabilities,
 			requiredCapabilities: requiredCapabilities
 		}).then(function (response) {
-			// At least geckodriver 0.15.0 returns the response data in a 'value' property, whereas Selenium does not.
-			if (response.value && response.value.sessionId) {
```
[the remark this answers is not in the document]
Comment (silne30) [reply Server.js]: @jason0x43 Fixed. I think we should be good now. Thanks for catching that.

**Comment by silne30** [comment]
@jason0x43 The edits have been made.

File: Server.js [hunk 2]
```diff
@@ -363,12 +363,22 @@ Server.prototype = {
...
-				response = response.value;
+			var responseData = null;
+			var sessionId = null;
+			// At least geckodriver 0.19.0 returns the response data in a 'value.capabilities'
+	        // property, whereas Selenium does not.
```
Comment (dylans) [inline Server.js]: Somehow my commit to fix the mixed use of tabs and spaces in this and the next few lines was lost with the recent push...

**Comment by silne30** [comment]
@dylans @jason0x43 I set webstorm up for smart tabs with 4 indentation. I hope this worked.

**Comment by jason0x43** [comment]
Merged in https://github.com/theintern/leadfoot/commit/b8618706fd3facda446661765b851eb075039ff2. Thanks!

**Pull request closed without merging**